### ISSUE TITLE
triple-slash reference directive: not working

### ISSUE THREAD

**Issue by mfulton26** [denoland/deno#10561]
Maybe I've misunderstood [providing types when hosting](https://deno.land/manual@v1.9.2/typescript/types#providing-types-when-hosting) but the following does not compile.

**a.d.ts**

```d.ts
declare const a: unique symbol;

declare global {
  interface Object {
    [a](): number;
  }
}

export default a;
```

**a.js**

```js
/// <reference path="./a.d.ts" />

const a = Symbol();

Object.defineProperty(Object.prototype, a, {
  value() {
    return 42;
  },
});

export default a;
```

**demo.ts**

```ts
import a from "./a.js";

console.log({}[a]());
//          ^
// Element implicitly has an 'any' type because expression of type 'unique symbol' can't be used to index type '{}'.
//   Property '[a]' does not exist on type '{}'.deno-ts(7053)
```

If I add `// @deno-types="./a.d.ts"` to the top of `demo.ts` then it compiles fine but I'd rather declare the association with the `d.ts` file somehow in the module instead of needing to do it in each import. Please advise. Thank you.

**Comment by lucacasonato** [comment]
This only doesn't work in the LSP it seems. Can you confirm `deno run` succeeds without issue?

**Comment by mfulton26** [comment]
Without `@deno-types` it fails to run. With it (or with `--no-check`) it runs successfully and prints `42`.

```
% deno run demo.ts
Check file:///path/to/demo.ts
error: TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'unique symbol' can't be used to index type '{}'.
  Property '[a]' does not exist on type '{}'.
console.log({}[a]());
            ~~~~~
    at file:///path/to/demo.ts:3:13
```

**Comment by kitsonk** [comment]
BAH! It is a typo on the [documentarion](https://deno.land/manual@v1.9.2/typescript/types#using-the-triple-slash-reference-directive) that no-one has caught. Does the following work?

```js
/// <reference types="./a.d.ts" />
```

**Comment by mfulton26** [comment]
Wow, yes, `deno run` succeeds without issue with `types` instead of `path`.

I am now seeing a different error from the language server though:

> Cannot find type definition file for './a.d.ts'.deno-ts(2688)

Does this need to be reported separately at https://github.com/denoland/vscode_deno?

**Comment by mfulton26** [comment]
Hmmm… shouldn't Deno be using `path` directives though? I just read the descriptions [here](https://www.staging-typescript.org/docs/handbook/triple-slash-directives.html) and it seems to me that `types` is used for _packages_ (e.g. for pulling in `"node"`) so I'm wondering if `path` is more appropriate and maybe an update to `deno` does not to be made (and then possibly none to `vscode_deno`).

**Comment by kitsonk** [comment]
> Does this need to be reported separately at https://github.com/denoland/vscode_deno?

No, because the language server is built into Deno.  I will take a look at it later.

> Hmmm… shouldn't Deno be using `path` directives though?

It is used in TypeScript files to do that.  Generally `tsc` pays no attention to what is in JavaScript files.  When we chose it, quite a long time ago, we acknowledge the risk of "squatting" on something in JavaScript files that `tsc` uses for another purpose in TypeScript files.
